### PR TITLE
mongodb Pauline Andersson

### PR DESCRIPTION
**Pull request by Pauan8**

### CODE REVIEW

File: Router/ticketmaster.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import express from 'express'
+import listEndpoints from 'express-list-endpoints'
+import mongoose from 'mongoose'
+
+import Events from '../data/ticketmaster.json'
+
+const router = express.Router();
```
Comment (rrpersson) [inline Router/ticketmaster.js:7]: well done for using Router

File: Router/ticketmaster.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        },
+        attractions: {
+          type: [{
+            name: String
+          }]
+        }
+      }
+    ]
+  }
+});
```
Comment (rrpersson) [inline Router/ticketmaster.js:80]: Bless javascript and all of it's curly brackets xD

File: Router/ticketmaster.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      resultArr = await Event.find()
+    } else { 
+      // filter by stored queries
+      resultArr = await Event.find({ $and: [...queryArr] }) 
+    }
+    return resultArr.length > 0 ? res.json(resultArr) : res.send({ result: "No events found. Try another search" })
+  } catch (err) {
+    res.status(404).send({ error: "Page not found" })
+  }
+});
```
Comment (rrpersson) [inline Router/ticketmaster.js:130]: Good job pushing results to an array

File: server.js [hunk 4]
```diff
@@ -1,38 +1,23 @@
...
+    ? next() 
+    : res.status(503).send({ Error: "No Connection to server" })
+}); 
+
+app.use('/', ticketmaster);
 
-// Start the server
 app.listen(port, () => {
   // eslint-disable-next-line
   console.log(`Server running on http://localhost:${port}`)
```
Comment (rrpersson) [inline server.js:23]: nice job keeping the server.js file clean!

**Review by rrpersson — COMMENTED** [review]
Nice little project and good job for using more than one file! The code looks good to me and it's structured in a way that makes it easy to follow and understand! :) Good job!!

File: Router/ticketmaster.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
+import express from 'express'
+import listEndpoints from 'express-list-endpoints'
+import mongoose from 'mongoose'
+
+import Events from '../data/ticketmaster.json'
+
+const router = express.Router();
+
```
Comment (KarolCislowski) [inline Router/ticketmaster.js:8]: When you already dissect it a little bit, maybe you will consider playing Abby Sciuto a little more and sending your Schemas to the separate classes in the `models` folder?

File: Router/ticketmaster.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    id: String,
+    url: String,
+    type: {
+      type: String
+    },
+    country: Object
+  }
+]);
+
+const EventSchema = mongoose.Schema({
```
Comment (KarolCislowski) [inline Router/ticketmaster.js:40]: Great, that you add more configuration to your fields than just the type!

File: Router/ticketmaster.js (hunk 3 above)
Comment (KarolCislowski) [inline Router/ticketmaster.js:130]: Sick filtering 😮

File: Router/ticketmaster.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    await Event.find({ "dates.start.dateTime": { $gte: startDate, $lte: endDate } })
+      .then((event) => {
+        return event ? res.json(event) : res.send({ result: "No events this summer 😞" })
+      })
+  } catch (err) {
+    res.status(404).send({ error: "Page not found" })
+  }
+})
+
+router.get('/events/id/:id', async (req, res) => {
```
Comment (KarolCislowski) [inline Router/ticketmaster.js:145]: `'/events/:id' ` when you are using the primary key to filter your data, do not specify its name in the endpoint 🤓

**Review by KarolCislowski — APPROVED** [review]
Great API 🌈 🦄

**Pull request closed without merging**